templates/data/third-party-images-data.yaml

```
@@ -365,3 +365,32 @@ templates:
                   global:
                     - "jobConfig_postsubmit"
                     - "kaniko_buildpack"
+
+              - jobConfig:
+                  name: "pre-main-tpi-otel-collector"
+                  run_if_changed: "^otel-collector/"
+                  args:
+                    - "--name=tpi/otel-collector"
+                    - "--config=/config/kaniko-build-config.yaml"
+                    - "--context=otel-collector"
+                    - "--dockerfile=Dockerfile"
+                    - "--env-file=envs"
+                inheritedConfigs:
+                  global:
+                    - "jobConfig_presubmit"
+                    - "kaniko_buildpack"
+              - jobConfig:
+                  name: "post-main-tpi-otel-collector"
+                  run_if_changed: "^otel-collector/"
+                  args:
+                    - "--name=tpi/otel-collector"
+                    - "--config=/config/kaniko-build-config.yaml"
+                    - "--context=otel-collector"
+                    - "--dockerfile=Dockerfile"
+                    - "--env-file=envs"
+                    - '--tag={{`{{ .Env \"KSM_VERSION\" }}-{{ .ShortSHA }}`}}'
```

You probably want your image tag in `OTEL_VERSION`-`SHA` format, as you provide `OTEL_VERSION` through envs file in thire-party-images repo
```suggestion
                    - '--tag={{`{{ .Env \"OTEL_VERSION\" }}-{{ .ShortSHA }}`}}'
```

---

thanks fixed